File: Baekjoon/S5/C#/S5_마트료시카 합치기.cs
https://www.acmicpc.net/problem/25631

using System;
using System.Collections.Generic;
using System.Linq;

namespace Beakjoon
{
    class Program
    {
        static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());
            int[] arr = Array.ConvertAll(Console.ReadLine().Split(' '), int.Parse);

            List<int> li = arr.ToList();
            li.Sort();

            if (li[0] == li[li.Count - 1])
            {
                Console.Write(li.Count);
                return;
            }

            int cnt = 1;
            int max = 0;
            while (li.Count > 1)
            {
                if (li[0] == li[1])
                    cnt++;
                else
                    cnt = 1;

                li.RemoveAt(0);
                max = Math.Max(max, cnt);
            }

            Console.Write(max);
        }
    }
}

File: Baekjoon/S5/JAVA/S5_마트료시카 합치기.java
//https://www.acmicpc.net/problem/25631

import java.io.*;
import java.util.*;

public class Main{
    public static void main(String args[]) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        
        StringTokenizer st = new StringTokenizer(br.readLine());
        int[] arr = new int[n];
        for (int i = 0; i < n; i++)
            arr[i] = Integer.parseInt(st.nextToken());
        
        Arrays.sort(arr);
        
		int max = 1;
		int cnt = 1;
		for(int i = 1; i < n; i++){
			if(arr[i] == arr[i-1])
			    cnt++;
			else
			    cnt = 1;
			    
			if(cnt > max)
			    max = cnt;
		}
        
        System.out.print(max);
    }
}